src/Core/Domain/ImageSettings/CommandHandler/RegenerateThumbnailsHandler.php

```diff
@@ -0,0 +1,113 @@
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\Exception\RegenerateThumbnailsTimeoutException;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\Exception\RegenerateThumbnailsWriteException;
+use PrestaShop\PrestaShop\Core\Language\LanguageRepositoryInterface;
+use PrestaShopBundle\Entity\Repository\ImageTypeRepository;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+#[AsCommandHandler]
+final class RegenerateThumbnailsHandler extends AbstractObjectModelHandler implements RegenerateThumbnailsHandlerInterface
+{
+    public function __construct(
+        private readonly TranslatorInterface $translator,
+        private readonly ImageTypeRepository $imageTypeRepository,
+        private readonly LanguageRepositoryInterface $langRepository,
+        private readonly ImageThumbnailsRegenerator $imageThumbnailsRegenerator,
+    ) {
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws ImageTypeException
+     */
+    public function handle(RegenerateThumbnailsCommand $command): void
+    {
+        // Get all languages
+        $languages = $this->langRepository->findAll();
+
+        // Set images to regenerate with all theirs specific directories
+        $process = [
+            ['type' => 'categories', 'dir' => _PS_CAT_IMG_DIR_],
+            ['type' => 'manufacturers', 'dir' => _PS_MANU_IMG_DIR_],
+            ['type' => 'suppliers', 'dir' => _PS_SUPP_IMG_DIR_],
+            ['type' => 'products', 'dir' => _PS_PRODUCT_IMG_DIR_],
+            ['type' => 'stores', 'dir' => _PS_STORE_IMG_DIR_],
+        ];
+
+        // Launching generation process
+        foreach ($process as $proc) {
+            // Check if this kind of image is selected, if not, skip
+            if ($command->getImage() !== 'all' && $command->getImage() !== $proc['type']) {
+                continue;
+            }
+
+            // Getting formats generation (all if 'all' selected)
+            if ($command->getImageTypeId() === 0) {
+                $formats = $this->imageTypeRepository->findAll();
+            } else {
+                $formats = $this->imageTypeRepository->findBy(['id' => $command->getImageTypeId()]);
+            }
+
+            // If user asked to erase images, let's do it first
+            if ($command->erasePreviousImages()) {
+                $this->imageThumbnailsRegenerator->deletePreviousImages($proc['dir'], $formats, $proc['type'] === 'products');
+            }
+
+            // Regenerate images
+            $errors = $this->imageThumbnailsRegenerator->regenerateNewImages($proc['dir'], $formats, $proc['type'] == 'products');
+            if (is_array($errors) && count($errors) > 0) {
+                if (in_array('timeout', $errors)) {
+                    throw new RegenerateThumbnailsTimeoutException($this->translator->trans('Only part of the images have been regenerated. The server timed out before finishing.', [], 'Admin.Design.Notification'));
```

Comment:
Just a question: when the timeout occurs, is it possible to re-run the action until it succeeds? Is the command able to catch up from where it timed out?